### DISCUSSION TITLE
REST API producer

### DISCUSSION THREAD

**Discussion by vicharerohan07** [AbsaOSS/spline-spark-agent#469] [General]
## Background [Optional]
I was trying to call "/Producer" Api from azure Databricks. and this /Producer api is written in C#. I want to send the data to my c# Application and not to any DB. i have install spline agent compatible with the Azure Databricks Runtime versions. but whenever i ran some query and and i am enabling the lineage tracking in python the it gives me error that za.co.absa.spline.harvester.exception.SplineInitializationException: Spark Agent was not able to establish connection to Spline Gateway. HTTP Response: 404

## Question
How i can hit the Custom producer API and send the spline data to it using spline agent from databricks. So that i can use it in my business logic ?

**Comment by wajda** [comment]
If I understand correctly you are using a default HTTP dispatcher. What REST endpoints have you implemented in your API? The dispatcher makes three calls:
```http
HEAD /status
POST /execution-plans
POST /execution-events
```
I also recommend to set the `spline.lineageDispatcher.http.apiVersion` property value to `LATEST` in the agent config, to receive the lineage data in the most up-to-date and detailed format. See [here](https://github.com/AbsaOSS/spline-spark-agent/blob/release/0.7/core/src/main/scala/za/co/absa/spline/harvester/dispatcher/HttpLineageDispatcher.scala#L58) for details about how the data model version negotiation works in the HTTP Dispatcher.
